### PR TITLE
docs: integrate experimental API into docs

### PR DESCRIPTION
**Pull request by zhenyasav**
Pls review within `react-client`:
- [ ] auto-redirect to halo within [useIdentity](https://github.com/dxos/dxos/pull/2397/files#diff-987a2a923f8ae8e26fa29a4d252adb81a01c7de98baeba46a8930433f192144aR13)
- [ ] grab first space with no arguments in [useSpace](https://github.com/dxos/dxos/pull/2397/files#diff-49ef141a581f5536c4aca81ab843b56b2ff0fa70db057629dc1e93a064e0a36fR16)
- [ ] support an undefined space in [useQuery](https://github.com/dxos/dxos/pull/2397/files#diff-ce4de75a52c8f72ff79916fd8d91fe6090d591eb22cf606e3d2d47e5d32d16f8R18) 

app
logged in

### CODE REVIEW

File: packages/sdk/react-client/src/echo/useQuery.ts [hunk 1]
```diff
@@ -8,24 +8,24 @@ import { Space } from '@dxos/client';
...
-  <T extends DocumentBase>(space: Space, filter?: Filter<T>): Document[];
+  <T extends DocumentBase>(space?: Space, filter?: TypeFilter<T>): T[];
+  <T extends DocumentBase>(space?: Space, filter?: Filter<T>): Document[];
 };
 
 /**
  * Create subscription.
  */
-export const useQuery: UseQuery = <T extends DocumentBase>(space: Space, filter?: Filter<T>): DocumentBase[] => {
+export const useQuery: UseQuery = <T extends DocumentBase>(space?: Space, filter?: Filter<T>): DocumentBase[] => {
```
Comment (wittjosiah) [inline packages/sdk/react-client/src/echo/useQuery.ts:18]: I considered doing this as well but was a little uneasy about returning an empty array when the space was undefined. If you think it's alright then I'm cool with it as it definitely makes the hook easier to work with.

File: packages/sdk/react-client/src/echo/useSpaces.ts [hunk 2]
```diff
@@ -13,10 +13,25 @@ import { useClient } from '../client';
...
+export const useSpace = (spaceKey?: PublicKeyLike | null, options?: { create?: boolean }) => {
+  const { create } = { create: false, ...options };
+  const client = useClient();
   const spaces = useSpaces();
-  const space = spaces.find((space) => spaceKey && space.key.equals(spaceKey));
-
+  const space = spaceKey ? spaces.find((space) => spaceKey && space.key.equals(spaceKey)) : spaces?.[0];
+  const creating = useRef(false);
+  useEffect(() => {
+    console.log('use space effect', space, create, !creating.current);
```
Comment (wittjosiah) [inline packages/sdk/react-client/src/echo/useSpaces.ts:23]: :eyes:

File: packages/sdk/react-client/src/echo/useSpaces.ts [hunk 3]
```diff
@@ -13,10 +13,25 @@ import { useClient } from '../client';
...
+    if (!space && create && !creating.current) {
+      creating.current = true;
+      client.echo
+        .createSpace()
+        .then(() => (creating.current = false))
+        .catch((err) => {
+          console.error(err);
+          creating.current = false;
+        });
```
[the remark this answers is not in the document]
Comment (zhenyasav) [reply packages/sdk/react-client/src/echo/useSpaces.ts:32]: aye

File: packages/sdk/react-client/src/echo/useQuery.ts (hunk 1 above)
Comment (zhenyasav) [reply packages/sdk/react-client/src/echo/useQuery.ts:18]: we'd do a suspense here if doing this properly right?

File: packages/sdk/react-client/src/echo/useSpaces.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhenyasav) [reply packages/sdk/react-client/src/echo/useSpaces.ts:32]: kinda wish we could cancel a promise yea? do we already have a `useAsyncEffect` thing?

File: packages/apps/quickstart/src/Counter.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+export const Counter = () => {
+  const identity = useIdentity({ login: true });
+  const space = useSpace(null, { create: true });
+  const [counter] = useQuery(space, { type: 'counter' });
+  const creating = useRef(false);
+  useEffect(() => {
+    console.log('counter effect', counter, space, creating.current);
+    if (!counter && space && !creating.current) {
+      creating.current = true;
+      const c = new Document({ type: 'counter' });
```
Comment (dmaretskyi) [inline packages/apps/quickstart/src/Counter.tsx:19]:
```suggestion
      const c = new Document({ type: 'counter', count: 0 });
```

File: packages/sdk/react-client/src/echo/useSpaces.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (wittjosiah) [reply packages/sdk/react-client/src/echo/useSpaces.ts:32]: I've been doing the following 
```ts
useEffect(() => {
  const timeout = setTimeout(async () => {
    // do stuff
  });
  
  return () => clearTimeout(timeout);
}, []);
```

File: packages/apps/quickstart/src/Counter.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+// Copyright 2023 DXOS.org
+//
+
+import React, { useEffect, useRef } from 'react';
+
+import { useQuery, Document, useIdentity, useSpace } from '@dxos/react-client';
+import { Loading } from '@dxos/react-components';
+
+export const Counter = () => {
+  const identity = useIdentity({ login: true });
```
Comment (dmaretskyi) [inline packages/apps/quickstart/src/Counter.tsx:11]:
```typescript
useEffect(() => {
  if(!space.database.root.counter) {
     space.database.root.counter = new Counter({ count: 0 })
   }
}, [space])

const counter = space.database.root.counter;

```

**Comment by zhenyasav** [comment]
check is passing, just stalling on cleaning up ssh sessions

**Pull request merged**